grails-app/services/org/pih/warehouse/inventory/InventoryService.groovy

```diff
@@ -3449,4 +3454,49 @@ class InventoryService implements ApplicationContextAware {
             ]
         }
     }
+
+    PaginatedList<ExpirationHistoryReportRow> getExpirationHistoryReport(ExpirationHistoryReportFilterCommand command) {
+        List<TransactionEntry> entries = TransactionEntry.createCriteria().list(offset: command.offset, max: command.max) {
+            transaction {
+                // Expired transaction type is hardcoded with id = "4"
+                eq("transactionType", TransactionType.read("4"))
+                eq("inventory", AuthService.currentLocation.inventory)
+                between("transactionDate", command.dateFrom, command.dateTo)
+                order("transactionDate", "desc")
+            }
+        }
+        return new PaginatedList<ExpirationHistoryReportRow>(entries.collect { ExpirationHistoryReportRow.buildExpirationReportRow(it) }, entries.totalCount)
```

Comment:
Given this is static, `ExpirationHistoryReportRow.buildExpirationReportRow(it)` seems like a needlessly repetitive name. How about `ExpirationHistoryReportRow.from(it)` or `ExpirationHistoryReportRow.of(it)`?